homeassistant/components/asterisk_mbox.py

```diff
@@ -41,41 +44,79 @@ def setup(hass, config):
     port = conf.get(CONF_PORT)
     password = conf.get(CONF_PASSWORD)
 
-    hass.data[DOMAIN] = AsteriskData(hass, host, port, password)
-
-    discovery.load_platform(hass, 'mailbox', DOMAIN, {}, config)
+    hass.data[DOMAIN] = AsteriskData(hass, host, port, password, config)
 
     return True
 
 
 class AsteriskData:
     """Store Asterisk mailbox data."""
 
-    def __init__(self, hass, host, port, password):
+    def __init__(self, hass, host, port, password, config):
         """Init the Asterisk data object."""
         from asterisk_mbox import Client as asteriskClient
 
         self.hass = hass
+        self.config = config
         self.client = asteriskClient(host, port, password, self.handle_data)
-        self.messages = []
+        self.messages = None
+        self.cdr = None
 
         async_dispatcher_connect(
             self.hass, SIGNAL_MESSAGE_REQUEST, self._request_messages)
+        async_dispatcher_connect(
```

Comment:
This must be called from within the event loop but it's called from `setup` ie from a worker thread. Change to use the sync version.